iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java

```diff
@@ -250,10 +250,14 @@ public void skipToGivenPosition(long pos) throws IOException {
         buffer.clear();
         channel.read(buffer);
         buffer.flip();
-        buffer.get();
+        CompressionType type = CompressionType.deserialize(buffer.get());
         currSegmentSize = buffer.getInt();
         if (posRemain >= currSegmentSize) {
           posRemain -= currSegmentSize;
+          channel.position(
+              channel.position()
+                  + currSegmentSize
+                  + (type == CompressionType.UNCOMPRESSED ? 0 : Integer.BYTES));
```

Comment:
You may skip with the channel by `currSegmentSize + Integer.BYTES` if the WAL is compressed, but `posRemain` is only reduced by `currSegmentSize`.
Is this inconsistency problematic?

Reply:
No. posRemain is a logical size, not a physical size. It represents the offset of the WAL before compression, so you only need to subtract the originalSegmentSize.

Reply:
This should be clarified by comments or better variable names; try it.

Reply:
I add a comment for it, and rename the function.